lib/action_controller/serialization.rb

```diff
@@ -27,6 +27,7 @@ def get_serializer(resource, options = {})
       end
       serializable_resource = ActiveModel::SerializableResource.new(resource, options)
       if serializable_resource.serializer?
+        serializable_resource.url_helper = self
```

Comment:
Sounds good. Would you be ok with this pr only including hash with proc links option? In the mean time it will allow to manually expose links and we can se for automatic link generator in the feature.